### PR TITLE
Bug 1868339: Move CredentialsRequest creation to CSO

### PR DESCRIPTION
**Pull request by bertinatto**
CSI operators should let CSO apply the CredentialsRequest objects, regardless of the cloud platform.

This patch is a requirement to allow users to manually see the secrets they need to create when running cloud-credentials-controller in Manual mode. More information at: https://github.com/openshift/cloud-credential-operator/blob/master/docs/mode-manual-creds.md.

Once this is merged, CredentialsRequests will be created by CVO, so we'll need follow up PRs to CSI operator repositories to **not create those objects themselves**.

Please see individual commits for a description of each change.

CC @openshift/storage

### CODE REVIEW

**Comment by bertinatto** [comment]
/hold

Before this merges I need to patch the CSI driver operators and then test manually.

**Comment by bertinatto** [comment]
Also, @dgoodwin FYI

**Comment by dgoodwin** [comment]
Thank you for this, looks good from my PoV.

**Comment by bertinatto** [comment]
/test e2e-aws

`Operator degraded (IngressControllersDegraded): Some ingresscontrollers are degraded: default`

**Review by tsmetana — APPROVED** [review]
/lgtm

**Comment by gnufied** [comment]
> Once this is merged, CredentialsRequests will be created by CVO, so we'll need follow up PRs to CSI operator repositories to not create those objects themselves.

@bertinatto Can CCO handle cases where secret was already created? Looks like it already does that otherwise this PR would have failed test I think.

**Comment by bertinatto** [comment]
> > Once this is merged, CredentialsRequests will be created by CVO, so we'll need follow up PRs to CSI operator repositories to not create those objects themselves.
> 
> @bertinatto Can CCO handle cases where secret was already created? Looks like it already does that otherwise this PR would have failed test I think.

Yes, from what I've seen it can handle that.

File: manifests/03_credentials_request_aws.yaml
Comment (gnufied) [inline manifests/03_credentials_request_aws.yaml:2]: should this be called "ebs-credential-request"? In case we have to create separate credential request for EFS for example?

File: manifests/03_credentials_request_aws.yaml
Comment (gnufied) [inline manifests/03_credentials_request_aws.yaml:9]: same applies to secret's name too.

Comment (bertinatto) [reply manifests/03_credentials_request_aws.yaml:9]: We'll need to have a coordinated way of changing the secret name here and in the CSI driver operator repo.

If I change it here, the CI job will likely fail because the CSI driver is looking for this secret name.

File: manifests/03_credentials_request_aws.yaml
Comment (bertinatto) [reply manifests/03_credentials_request_aws.yaml:2]: Changed to `aws-ebs-csi-driver-operator` to align with the other CCs which are using the operator name.

**Comment by bertinatto** [comment]
/hold cancel

CSI driver operators should be changed after this PR merges.

File: manifests/03_credentials_request_aws.yaml
Comment (bertinatto) [reply manifests/03_credentials_request_aws.yaml:9]: Renamed to `ebs-cloud-credentials` (similar to Manila and oVirt).

It shouldn't fail CI because for the time being the CSI driver operator is creating the secret itself.

**Pull request merged**
